arduino/smartcar/smartcar.ino

```diff
@@ -1,9 +1,41 @@
 #include <Smartcar.h>
 
-void setup() {
-  // put your setup code here, to run once:
+const int TRIGGER_PIN           = 6; 
+const int ECHO_PIN              = 7; 
+const unsigned int MAX_DISTANCE = 200;
+const int FRONT_PIN = 0;
+
+ArduinoRuntime arduinoRuntime;
+SR04 front(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+GP2Y0A21 frontIR(arduinoRuntime, FRONT_PIN); 
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+
+SimpleCar car(control);
+
+void setup()
+{
+    Serial.begin(9600);
+    car.setSpeed(30);
 }
 
-void loop() {
-  // put your main code here, to run repeatedly:
+void loop()
+{
+   int usDistance = front.getDistance();
+   int irDistance = frontIR.getDistance();
+  if((usDistance < 170) && (usDistance > 30) )
```

Comment:
What happens if `usDistance` is `0`, i.e. an error (no obstacle found)?

Reply:
If there is no obstacle the car will just continue forward and if the distance is less than 30 then the IR sensor stops the car.

Reply:
I guess you are using the other sensor for smaller distances so that will work, OK